packages/kit/test/apps/basics/test/client.test.js

```diff
@@ -1254,3 +1254,45 @@ test.describe('Interactivity', () => {
 		expect(errored).toBe(false);
 	});
 });
+
+test.describe('+loading.svelte', () => {
+	test('shows loading ui', async ({ page, clicknav }) => {
+		await page.goto('/loading-ui');
+		clicknav('.client-load-1'); // do not await!
+		await page.waitForTimeout(100); // give time to rerender
+		expect(await page.textContent('h3')).toBe('Loading');
+		await page.waitForTimeout(1200);
+		expect(await page.textContent('h3')).toBe('Result');
+	});
```

Comment:
I think `waitForTimeout` are not necessary here and can be avoided. `timeout` option can be passed to `textContent` if needed.